backend/app/api/transactions.py

```
@@ -115,14 +125,45 @@ async def import_transactions(
+                        # For traditional assets, try to fetch ISIN from Yahoo Finance
+                        try:
+                            fetched_isin, fetched_description = await fetch_ticker_metadata(ticker)
+                            isin = fetched_isin
+                            if fetched_description and not description:
+                                description = fetched_description
+                            logger.info(f"Fetched ISIN {isin} for ticker {ticker}")
+                        except Exception as e:
+                            logger.warning(f"Could not fetch ISIN for {ticker}: {str(e)}")
+                            # For traditional assets without ISIN, create a 12-char placeholder
+                            # Generate deterministic hash-based ISIN
+                            ticker_hash = hashlib.sha1(ticker.upper().encode()).hexdigest()[:10].upper()
```

Ruff (0.13.3)

Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)